### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by scharlottej13**
- [ ] partially addresses #5445

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: .coveragerc [hunk 1]
```diff
@@ -11,9 +11,21 @@ omit =
...
+    distributed/pytest_resourceleaks.py
+    distributed/comm/ucx.py
 
 [report]
 show_missing = True
+exclude_lines =
+    # re-enable the standard pragma
+    pragma: no cover
+    # exclude nvml calls
+    \snvml\.
```
Comment (scharlottej13) [inline .coveragerc:25]: I think this should actually be something like `[\s(.]nvml\.` (and same for pynvml)

**Comment by scharlottej13** [comment]
Here's a comparison of all files with lines excluded from my local html coverage report, since I didn't see an equivalent from codecov.io (but maybe there's a better way to do this!)

Module | excluded (main) | excluded (test-coverage)
-- | -- | -- 
distributed/scheduler.py | 2 | 70 
distributed/nanny.py | 67 | 67 
distributed/worker.py | 0 | 28
distributed/diagnostics/nvml.py | 0 | 22
distributed/system_monitor.py | 0 | 15
distributed/stealing.py | 0 | 7
distributed/active_memory_manager.py | 5 | 5
distributed/core.py | 0 | 4
distributed/dashboard/scheduler.py | 0 | 2
distributed/dashboard/components/nvml.py | 0 | 1

**Comment by fjetter** [comment]
I'm wondering if we should adjust our target to `90%` (what we have right now) or `auto` (such that it has to increase). Similarly, I'm wondering if we shouldn't set [patch status](https://docs.codecov.com/docs/commit-status#patch-status) reports to give better feedback about the code changes of a PR instead of only global coverage

cc @crusaderky @jrbourbeau 

I personally would also be interested in trying out [github checks](https://docs.codecov.com/docs/github-checks) ([config](https://docs.codecov.com/docs/codecovyml-reference#github_checks-github-users-only)) but we can defer this to a later iteration

**Comment by bryanwweber** [comment]
> I'm wondering if we should adjust our target to `90%` (what we have right now) or `auto` (such that it has to increase).

@fjetter The only disadvantage of `auto` that I've found in other projects is that some PRs deliberately do not address coverage for one reason or another (including that Codecov has mis-measured coverage). So we might need to be prepared to merge PRs with this check "failing". But that might be a "deal with it later" problem.

> Similarly, I'm wondering if we shouldn't set [patch status](https://docs.codecov.com/docs/commit-status#patch-status) reports to give better feedback about the code changes of a PR instead of only global coverage

In other projects I've found these status reports to be very useful, if occasionally noisy when Codecov doesn't update coverage correctly. Although that might be because the project I'm thinking of is a combined C++/Cython project

> I personally would also be interested in trying out [github checks](https://docs.codecov.com/docs/github-checks) ([config](https://docs.codecov.com/docs/codecovyml-reference#github_checks-github-users-only)) but we can defer this to a later iteration

Again from experience, I personally find these line-by-line comments to be somewhat annoying because they tend to get in the way of review comments, but that might be related to the nature of the other projects and the (lack of) coverage they have.

File: .coveragerc [hunk 2]
```diff
@@ -11,9 +11,21 @@ omit =
     distributed/utils_test.py
     distributed/deploy/ssh.py
     distributed/_ipython_utils.py
+    distributed/_version.py
```
Comment (crusaderky) [inline .coveragerc:14]: I can spot a bunch of files that don't exist anymore above this line - could you clean them up?

**Comment by crusaderky** [comment]
> * [ ]  closes [Misleading test coverage report #5445](https://github.com/dask/distributed/issues/5445)

I don't think this PR closes the issue; specifically, it's missing:
> - We have cases with relatively long switch-like statements ending in a final exception, e.g. ValueError "unknown value"
> - Conditional imports may need be covered with pragma statements and/or a dedicated job
> - All very obvious cases should have a pragma statement and/or exclude configuration option
> - Developer documentation is updated with a guideline

It's OK if the above points happen in a later PR, but the reference issue should remain open until then.

**Comment by crusaderky** [comment]
Screenshot from the codecov report below, which makes me worried:
Screenshot from 2022-01-25 14-38-54

Clearly the PR has no actual impact on it. Why is this happening? Is it because a flaky test is failing before it reaches the line? Or is it highlighting a line that is randomly sometimes hit and sometimes not? In either case, if people need to start reading the coverage reports, this kind of noise is going to become a problem as bad as the flaky tests.

**Comment by crusaderky** [comment]
> I'm wondering if we should adjust our target to `90%` (what we have right now) or `auto` (such that it has to increase). Similarly, I'm wondering if we shouldn't set [patch status](https://docs.codecov.com/docs/commit-status#patch-status) reports to give better feedback about the code changes of a PR instead of only global coverage

Not sure about patch status reports, but inspecting the current codecov report for this PR took me an uncomfortable amount of clicking. It would be nice to have a single-page view of everything I missed. Or not - see my post above about what is clearly a high degree of noise.

File: .coveragerc (hunk 2 above)
Comment (scharlottej13) [reply .coveragerc:14]: absolutely! thanks for spotting this.

**Comment by scharlottej13** [comment]
> Not sure about patch status reports, but inspecting the current codecov report for this PR took me an uncomfortable amount of clicking. It would be nice to have a single-page view of everything I missed. Or not - see my post above about what is clearly a high degree of noise.

If I'm understanding you correctly, this is something I was wondering about as well (and why the PR sat as a draft for a bit). The way I checked if a regex I added was working was clicking through the individual html reports (still a lot of clicking). It looks like there's also [`diff-cover`](https://diff-cover.readthedocs.io/en/latest/README.html#), but it didn't quite work as I expected when I tried it. I can look into this more, though, to see if I can get it to work.

**Comment by crusaderky** [comment]
LGTM! 
As I feared, utils_test.py needs some attention - something to be discussed at the next standup.

**Pull request merged**